source/usage-examples/changeStream.txt

```diff
@@ -7,76 +7,84 @@ Watch for Changes
+Open a change stream by calling the ``watch()`` method on a ``Collection``, ``Db``, or ``MongoClient`` object. The
+change stream emits **change event** documents when they occur.
 
-The ``watch()`` method optionally takes an **aggregation pipeline**  which
-consists of an array of **stages** as the first parameter to filter and
-transform the change events output as follows:. 
+The ``watch()`` method optionally takes an **aggregation pipeline** which consists of an array of **aggregation stages**
+as the first parameter. The aggregation stages filter and transform the change events.
+
+In the example below, the ``$match`` stage will match all change event documents with a ``runtime`` value of less than
+15, filtering all others out.
 
 .. code-block:: javascript
 
-   const pipeline = [ { $match: { runtime: { $lt: 15 } }, ];
+   const pipeline = [ { $match: { runtime: { $lt: 15 } } ];
    const changeStream = await collection.watch(pipeline);
 
-The ``watch()`` method accepts an additional ``options`` object as the second
-parameter. Refer to the links at the end of this section for more
-information on the settings you can configure in this object.
+The ``watch()`` method accepts an ``options`` object as the second parameter. Refer to the links at the end of this
+section for more information on the settings you can configure with this object.
 
-The ``watch()`` method returns an instance of a ``ChangeStream``. You can
-call methods on the ChangeStream such as ``hasNext()`` to check for remaining
-documents in the stream, ``next()`` to request the next document in the
-stream, ``pause()`` to stop emitting events, ``resume()`` to continue to
-emit events, and ``close()`` to close the ChangeStream. You can also attach
-listener functions by calling the ``on()`` method on the instance. See the
-link to the ``ChangeStream`` API documentation below for more details on the
-available methods.
+The ``watch()`` method returns an instance of a :node-api:`ChangeStream </ChangeStream.html>`. You can read events from
+change streams by iterating over them or listening for events. Select the tab that corresponds to the way you want to
+read events from the change stream below.
 
-Visit the following resources for additional material on the classes and
-methods presented above:
+.. warning::
 
-- :manual:`change streams </changeStreams/>`
-- :manual:`change events </reference/change-events/>`
-- :manual:`aggregation pipeline </reference/operator/aggregation-pipeline/>`
-- :manual:`aggregation stages </changeStreams/#modify-change-stream-output>`
-- :node-api:`ChangeStream class API documentation <api/ChangeStream.html>`
-- :node-api:`Collection.watch() </Collection.html#watch>`,
-  :node-api:`Db.watch() </Db.html#watch>`,
-  :node-api:`MongoClient.watch() API documentation </MongoClient.html#watch>`
+   Avoid mixing both the iterative and event-based approach of reading from change streams as it can cause unexpected
+   behavior.
```

Comment:
Is it worth explaining this a bit more? Like, what happens?

Reply:
forgot to rebase, this content isn't under review